worlds/stardew_valley/logic/crafting_logic.py

```diff
@@ -54,8 +54,8 @@ def knows_recipe(self, recipe: CraftingRecipe) -> StardewRule:
                 return self.logic.crafting.received_recipe(recipe.item)
         if self.options.craftsanity == Craftsanity.option_none:
             return self.logic.crafting.can_learn_recipe(recipe)
-        if isinstance(recipe.source, StarterSource) or isinstance(recipe.source, ShopTradeSource) or isinstance(
-                recipe.source, ShopSource):
+        if isinstance(recipe.source, StarterSource) or isinstance(recipe.source, ShopTradeSource) or\
+                isinstance(recipe.source, ShopSource) or isinstance(recipe.source, SkillCraftsanitySource):

```

Comment:
`isinstance()` can take a tuple of types as the second argument, returning `True` if the object is an instance of any of the types.
```suggestion
        if isinstance(recipe.source, (StarterSource, ShopTradeSource, ShopSource, SkillCraftsanitySource)):
```

Reply:
Oh I didn't know that. Thanks!